pkg/server/plugin/upstreamauthority/spire/spire.go

```diff
@@ -206,10 +206,8 @@ func (m *Plugin) pollBundleUpdates(ctx context.Context) {
 		resp, err := m.serverClient.getBundle(ctx)
 		if err != nil {
 			m.log.Warn("Failed to fetch bundle while polling", "error", err)
-		}
-
-		if m.bundleVersion == preFetchCallVersion && resp != nil {
-			m.setBundle(resp)
+		} else {
+			m.setBundleIfVersionMatches(resp, preFetchCallVersion)
```

Comment:
This is the fix to the race in the production code. The rest applies to the unit-tests.